src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -94,20 +94,33 @@ public void changeMemberPosition(Long memberId, Long queueId,
 
   @Override
   @Transactional(isolation = Isolation.REPEATABLE_READ)
-  public void deleteMember(Long memberId, Long queueId, String principalName) {
+  public void deleteMemberByCreatorOrManager(Long memberId, Long queueId, String principalName) {
     if (isUserQueueCreatorOrManagerByCredentials(queueId, principalName)) {
-      var entry = entryRepository.findById(new EntryId(memberId, queueId))
-          .orElseThrow(() -> new EntryNotFoundException(
-              "Could not find entry by memberId %s and queueId %s".formatted(memberId, queueId)));
-      entryRepository.deleteById(entry.getId());
-      entryRepository.updateEntryIndices(queueId, entry.getEntryIndex() + 1,
-          Integer.MAX_VALUE, -1);
+      deleteMember(memberId, queueId);
     } else {
       throw new ForbiddenAccessException(
           "User %s is not a creator or manager".formatted(principalName));
     }
   }
 
+  @Override
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
+  public void leaveByMember(Long queueId, String principalName) {
+    var user = userRepository.findByUsername(principalName)
+        .orElseThrow(() -> new UserNotFoundException(
+            "No user exists by given username: %s".formatted(principalName)));
+    deleteMember(user.getId(), queueId);
+  }
+
+  private void deleteMember(Long memberId, Long queueId) {
+    var entry = entryRepository.findById(new EntryId(memberId, queueId))
+        .orElseThrow(() -> new EntryNotFoundException(
+            "Could not find entry by memberId %s and queueId %s".formatted(memberId, queueId)));
+    entryRepository.deleteById(entry.getId());
+    entryRepository.updateEntryIndices(queueId, entry.getEntryIndex() + 1,
+        Integer.MAX_VALUE, -1);
```

Comment:
Imho, Passing Integer.MAX_VALUE as a method argument looks like an indicator that smth is wrong